File: python/stockservice/stock.py
class Stock(object):
    """
    A structured representation of stock information, including ticker symbol, latest sale price, and price change since yesterday.
    """
    def __init__(self, id, ticker, exchange, last, last_trade_date, last_trade_time, change, percent_change):
        """
        Creates a new Stock
        
        :param self: This object
        :type self: Stock
        :param id: The unique ID number for this ticker symbol
        :type id: int
        :param ticker: The Ticker Symbol (e.g. AAPL)
        :type ticker: string
        :param exchange: The name of the exchange (e.g. NASDAQ)
        :type exchange: string
        :param last: The latest sale price for this stock.
        :type last: float
        :param last_trade_date: The entire date of the last trade.
        :type last_trade_date: string
        :param last_trade_time: The time of the last trade.
        :type last_trade_time: string
        :param change: The price change since yesterday.
        :type change: float
        :param percent_change: The percent price change since yesterday.
        :type percent_change: float
        :returns: Stock
        """
        self.id = id
        self.ticker = ticker
        self.exchange = exchange
        self.last = last
        self.last_trade_date = last_trade_date
        self.last_trade_time = last_trade_time
        self.change = change
        self.percent_change = percent_change
    
    @staticmethod
    def _from_json(json_data):
        """
        Creates a Stock from json data.
        
        :param json_data: The raw json data to parse
        :type json_data: dict
        :returns: Stock
        """
        return Stock(json_data['id'],
                    json_data['t'],
                    json_data['e'],
                    json_data['l'],
                    json_data['lt'],
                    json_data['ltt'],
                    json_data['c'],
                    json_data['cp'])

File: java/src/realtimeweb/stockservice/domain/Stock.java
package realtimeweb.stockservice.domain;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Iterator;
import java.util.List;
import java.util.Map;




/**
 * A structured representation of stock information, including ticker symbol, latest sale price, and price change since yesterday.
 */
public class Stock {
	
    private Double percent_Change;
    private String exchange;
    private String last_Trade_Date;
    private String last_Trade_Time;
    private Double last;
    private String ticker;
    private Long id;
    private Double change;
    
    
    /*
     * @return The percent price change since yesterday.
     */
    public Double getPercent_Change() {
        return this.percent_Change;
    }
    
    /*
     * @param The percent price change since yesterday.
     * @return Double
     */
    public void setPercent_Change(Double percent_Change) {
        this.percent_Change = percent_Change;
    }
    
    /*
     * @return The name of the exchange (e.g. NASDAQ)
     */
    public String getExchange() {
        return this.exchange;
    }
    
    /*
     * @param The name of the exchange (e.g. NASDAQ)
     * @return String
     */
    public void setExchange(String exchange) {
        this.exchange = exchange;
    }
    
    /*
     * @return The entire date of the last trade.
     */
    public String getLast_Trade_Date() {
        return this.last_Trade_Date;
    }
    
    /*
     * @param The entire date of the last trade.
     * @return String
     */
    public void setLast_Trade_Date(String last_Trade_Date) {
        this.last_Trade_Date = last_Trade_Date;
    }
    
    /*
     * @return The time of the last trade.
     */
    public String getLast_Trade_Time() {
        return this.last_Trade_Time;
    }
    
    /*
     * @param The time of the last trade.
     * @return String
     */
    public void setLast_Trade_Time(String last_Trade_Time) {
        this.last_Trade_Time = last_Trade_Time;
    }
    
    /*
     * @return The latest sale price for this stock.
     */
    public Double getLast() {
        return this.last;
    }
    
    /*
     * @param The latest sale price for this stock.
     * @return Double
     */
    public void setLast(Double last) {
        this.last = last;
    }
    
    /*
     * @return The Ticker Symbol (e.g. AAPL)
     */
    public String getTicker() {
        return this.ticker;
    }
    
    /*
     * @param The Ticker Symbol (e.g. AAPL)
     * @return String
     */
    public void setTicker(String ticker) {
        this.ticker = ticker;
    }
    
    /*
     * @return The unique ID number for this ticker symbol
     */
    public Long getId() {
        return this.id;
    }
    
    /*
     * @param The unique ID number for this ticker symbol
     * @return Integer
     */
    public void setId(Long id) {
        this.id = id;
    }
    
    /*
     * @return The price change since yesterday.
     */
    public Double getChange() {
        return this.change;
    }
    
    /*
     * @param The price change since yesterday.
     * @return Double
     */
    public void setChange(Double change) {
        this.change = change;
    }
    
	
	/**
	 * Creates a string based representation of this Stock.
	
	 * @return String
	 */
	public String toString() {
		return "Stock[" +percent_Change+", "+exchange+", "+last_Trade_Date+", "+last_Trade_Time+", "+last+", "+ticker+", "+id+", "+change+"]";
	}
	
	/**
	 * Internal constructor to create a Stock from a json representation.
	 * @param map The raw json data that will be parsed.
	 * @return 
	 */
    public Stock(Map<String, Object> raw) {
        // TODO: Check that the data has the correct schema.
        // NOTE: It's much safer to check the Map for fields than to catch a runtime exception.
        try {
            this.percent_Change = Double.parseDouble(raw.get("cp").toString());
            this.exchange = raw.get("e").toString();
            this.last_Trade_Date = raw.get("lt").toString();
            this.last_Trade_Time = raw.get("ltt").toString();
            this.last = Double.parseDouble(raw.get("l").toString());
            this.ticker = raw.get("t").toString();
            this.id = Long.parseLong(raw.get("id").toString());
            this.change = Double.parseDouble(raw.get("c").toString());
        } catch (NullPointerException e) {
    		System.err.println("Could not convert the response to a Stock; a field was missing.");
    		e.printStackTrace();
    	} catch (ClassCastException e) {
    		System.err.println("Could not convert the response to a Stock; a field had the wrong structure.");
    		e.printStackTrace();
        }
    
	}	
}